File: system-test/dotnet/SystemTests/Legacy/Mod11/ExternalSystemContractTests/Clock/BaseClockContractTest.cs
using SystemTests.Legacy.Mod11.ExternalSystemContractTests.Base;

namespace SystemTests.Legacy.Mod11.ExternalSystemContractTests.Clock;

public abstract class BaseClockContractTest : BaseExternalSystemContractTest
{
    [Fact]
    public async Task ShouldBeAbleToGetTime()
    {
        (await Scenario()
            .Given().Clock().WithTime("2024-01-02T09:00:00Z")
            .Then().Clock())
            .HasTime();
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod11/contract/clock/BaseClockContractTest.java
package com.optivem.shop.systemtest.legacy.mod11.contract.clock;

import com.optivem.shop.systemtest.legacy.mod11.contract.base.BaseExternalSystemContractTest;
import org.junit.jupiter.api.Test;

public abstract class BaseClockContractTest extends BaseExternalSystemContractTest {
    @Test
    void shouldBeAbleToGetTime() {
        scenario
                .given().clock().withTime()
                .then().clock().hasTime();
    }
}
